tools/topology/sof/pipe-volume-demux-playback.m4

```diff
@@ -22,8 +22,29 @@ include(`muxdemux.m4')
 include(`mixercontrol.m4')
 include(`bytecontrol.m4')
 
-# Use default parameters
-include(`demux_coef_default.m4')
+dnl pipeline_id, channels, routing_matrix_rows
+define(matrix1, `ROUTE_MATRIX(PIPELINE_ID, 2,
+			     `BITS_TO_BYTE(1, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 1, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 1 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,1 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,1 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,1 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,1 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,1)')')
+
+define(matrix2, `ROUTE_MATRIX(5, 2,
```

Comment:
can we make this 5 as an DEFINE value here? So we can override it with some define when include this file.
Same for the 2 here, it should use channel variable.

Reply:
@xiuli Artur's PR #2600 will make these fields obsolete, so we should not anymore need to define these at all. I need to remove these after that PR is merged.

Reply:
moved the demux configuration to upper level so pipiline_id's are visible in the same file. You could also use defines. Generally I'm not huge fan of having defines in upper level topology files and using them in lower level building blocks. Makes understanding the structure really difficult...